Sources/XcodeProjMapper/Mappers/Workspace/XCWorkspaceMapper.swift

```diff
@@ -0,0 +1,125 @@
+            renderMarkdownReadme: false
+        )
+
+        return Workspace(
+            path: srcPath,
+            xcWorkspacePath: xcWorkspacePath,
+            name: workspaceName,
+            projects: projectPaths,
+            schemes: schemes,
+            generationOptions: generationOptions,
+            ideTemplateMacros: nil,
+            additionalFiles: []
+        )
+    }
+
+    // MARK: - Private Helpers
+
+    /// Recursively identifies all `.xcodeproj` files within the workspace structure.
+    ///
+    /// - Parameters:
+    ///   - elements: The workspace elements (files/groups).
+    ///   - srcPath: The source directory path used for resolving relative references.
+    /// - Returns: An array of absolute paths to `.xcodeproj` directories.
+    private func extractProjectPaths(
+        from elements: [XCWorkspaceDataElement],
+        srcPath: AbsolutePath,
+        workspaceProvider: WorkspaceProviding
+    ) throws -> [AbsolutePath] {
+        var paths = [AbsolutePath]()
+
+        for element in elements {
+            switch element {
+            case let .file(ref):
+                let refPath = try ref.path(srcPath: srcPath)
+                if refPath.fileExtension == .xcodeproj {
+                    paths.append(refPath)
+                }
+            case let .group(group):
+                let nestedSrcPath = srcPath.appending(component: group.location.path)
+                let groupPaths = try extractProjectPaths(
+                    from: group.children,
+                    srcPath: nestedSrcPath,
+                    workspaceProvider: workspaceProvider
+                )
+                paths.append(contentsOf: groupPaths)
+            }
+        }
+
+        return paths
+    }
+
+    /// Maps shared schemes defined within the workspace.
+    ///
+    /// Schemes are typically located in `xcshareddata/xcschemes`. If found,
+    /// this method parses them and maps them into `Scheme` models.
+    ///
+    /// - Parameter srcPath: The workspace's root path.
+    /// - Returns: An array of `Scheme` instances for shared schemes in the workspace.
+    private func mapSchemes(
+        from srcPath: AbsolutePath,
+        workspaceProvider: WorkspaceProviding
+    ) throws -> [Scheme] {
+        var schemes = [Scheme]()
+        let sharedDataPath = Path(srcPath.pathString) + "xcshareddata/xcschemes"
+
+        if sharedDataPath.exists {
+            let schemePaths = try sharedDataPath.children().filter { $0.extension == "xcscheme" }
+
+            // Construct graphType for schemes
+            let graphType = GraphType.workspace(workspaceProvider)
+            let schemeMapper = XCSchemeMapper()
+            for schemePath in schemePaths {
+                let xcscheme = try XCScheme(path: schemePath)
+                let scheme = try schemeMapper.map(xcscheme, shared: true, graphType: graphType)
+                schemes.append(scheme)
+            }
+        }
+
+        return schemes
```

Comment:
nit: I generally prefer not using `var`s if not necessary as it makes it easier to spot where the final values are coming from:
```suggestion
        let sharedDataPath = Path(srcPath.pathString) + "xcshareddata/xcschemes"

        if sharedDataPath.exists {
            let schemePaths = try sharedDataPath.children().filter { $0.extension == "xcscheme" }

            try schemePaths.map { schemePath in
                try schemeMapper.map(
                  try XCScheme(path: schemePath), 
                  shared: true, 
                  graphType: .workspace(workspaceProvider)
                )
            }
        } else {
          return []
        }
```

Note I removed `schemeMapper` initialization as that belongs into the `init` of `XCWorkspaceMapper` as it should be dependency injected